crates/sui-core/src/unit_tests/submit_transaction_tests.rs

```diff
@@ -169,11 +171,13 @@ async fn test_submit_transaction_already_executed() {
 
     // Verify we still got a consensus position back, because the transaction has not been committed yet,
     // so we can still sign the same transaction.
-    let response1 = SubmitTxResponse {
-        consensus_position: ConsensusPosition::try_from(response1.consensus_position.as_ref())
-            .unwrap(),
+    let response1: SubmitTxResponse = response1.try_into().unwrap();
```

Comment:
Shouldn't the end of this test now return the effects when you call submit transaction again instead of throwing an error? Also would be good to add a test that shows that "resubmission" works across epochs as we will just return the effects